OrleansDashboard/Implementation/Grains/DashboardGrain.cs

```diff
@@ -249,5 +254,177 @@ public Task SubmitTracing(string siloAddress, Immutable<SiloGrainTraceEntry[]> g
 
             return Task.CompletedTask;
         }
+
+        public async Task<Immutable<string>> GetGrainState(string id, string grainType)
+        {
+            var result = new ExpandoObject();
+
+            try
+            {
+                var _grainType = grainType.Split(".").Last();
+
+
+                var implementationType = AppDomain.CurrentDomain.GetAssemblies()
+                                    .SelectMany(s => s.GetTypes())
+                                    .Where(w => w.Name.Equals(_grainType))
+                                    .FirstOrDefault();
+
+                object grainId = null;
+                string keyExtension = "";
+                var splitedGrainId = id.Split(",");
```

Comment:
comma separation makes me slightly nervous, as people might use commas in the key names. For now we can live with this...

Reply:
ok